lib/os/thread_posix.cpp

```diff
@@ -70,29 +66,34 @@ namespace pstore {
 #        elif defined(PSTORE_PTHREAD_SET_NAME_NP)
             pthread_set_name_np (pthread_self (), name);
 #        else
-#            error "pthread_setname was not available"
+            std::strncpy(thread_name, name, name_size);
+            thread_name[name_size - 1] = '\0';
```

Comment:
I’m concerned about a possible mismatch here. If none of the pthread_setname() APIs that we know about are available, but the getname() API _is_, then we’ll set the name using a fallback but get it using the API. That wouldn’t be right.

(Don’t forget to clang-format!)

Reply:
Good point. Fixed in the commit https://github.com/SNSystems/pstore/pull/93/commits/88195a24bcbd4e6e349f08bc63cdc1197b10f153.